lib/simple_forms_api/form_remediation/configuration/base.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+module SimpleFormsApi
+  module FormRemediation
+    module Configuration
+      class Base
+        attr_reader :id_type, :include_manifest, :include_metadata, :parent_dir, :presign_s3_url
+
+        def initialize
+          @id_type = :benefits_intake_uuid  # The field to query the FormSubmission by
+          @include_manifest = true          # Include a CSV file containing manifest data
+          @include_metadata = false         # Include a JSON file containing form submission metadata
+          @parent_dir = '/'                 # The base directory in the S3 bucket where the archive will be stored
+          @presign_s3_url = true            # Once archived to S3, the service should generate & return a presigned_url
+        end
+
+        # Override to inject your team's own submission archive
+        def submission_archive_class
+          SimpleFormsApi::FormRemediation::SubmissionArchive
+        end
+
+        # Override to inject your team's own s3 client
+        def s3_client
+          SimpleFormsApi::FormRemediation::S3Client
+        end
+
+        # Override to inject your team's own submission data builder
+        def remediation_data_class
+          SimpleFormsApi::FormRemediation::SubmissionRemediationData
+        end
+
+        # Override to inject your team's own file uploader
+        # If overriding this, s3_settings method doesn't have to be set
+        def uploader_class
+          SimpleFormsApi::FormRemediation::Uploader
+        end
+
+        # The FormSubmission model to query against
+        def submission_type
+          FormSubmission
+        end
+
+        # The attachment model to query for form submission attachments
+        def attachment_type
+          PersistentAttachment
+        end
+
+        # The temporary directory where form submissions will be
+        # hydrated and stored. This directory will automatically
+        # be deleted once the archive process completes
+        def temp_directory_path
+          @temp_directory_path ||= Rails.root.join("tmp/#{SecureRandom.hex}-archive/").to_s
```

Comment:
How long are submissions stored in the tmp directory? Just a reminder that pods are ephemeral and the tmp dir is only alive as long as the lifecycle of a pod

Reply:
Only long enough for the data contained within to be copied over to S3. This is purely just a working space to gather, zip, and then hand off to the S3 client.

Reply:
Ok, so were talking just a matter of ms or s?

Reply:
I'd say ms.